main/src/ascent_adaptor.h

```diff
@@ -103,41 +104,59 @@ void addField(conduit::Node& mesh, const std::string& name, FieldType* field, si
     mesh["fields/" + name + "/volume_dependent"].set("false");
 }
 
-template<class DataType>
-void Execute(DataType& d, long startIndex, long endIndex)
+template<class DataType, class DomainType, class ParticleDataType>
+void Execute(DataType& d, std::unique_ptr<sphexa::Propagator<DomainType, ParticleDataType>>& p, long startIndex,
+             long endIndex, size_t rank)
 {
     conduit::Node mesh;
     mesh["state/cycle"].set_external(&d.iteration);
     mesh["state/time"].set_external(&d.ttot);
+    mesh["state/domain_id"] = rank;
+
+    // The execution time for 1 iteration in current rank. Differs upon ranks
+    float currTimeDelta = p->getTotalIterationTime();
+    mesh["state/time"].set(currTimeDelta);
 
     mesh["coordsets/coords/type"] = "explicit";
     mesh["coordsets/coords/values/x"].set_external(&d.x[startIndex], endIndex - startIndex);
     mesh["coordsets/coords/values/y"].set_external(&d.y[startIndex], endIndex - startIndex);
     mesh["coordsets/coords/values/z"].set_external(&d.z[startIndex], endIndex - startIndex);
 
-    mesh["topologies/mesh/type"]     = "unstructured";
-    mesh["topologies/mesh/coordset"] = "coords";
-
-    addField(mesh, "x", d.x.data(), startIndex, endIndex);
-    addField(mesh, "y", d.y.data(), startIndex, endIndex);
-    addField(mesh, "z", d.z.data(), startIndex, endIndex);
-    addField(mesh, "vx", d.vx.data(), startIndex, endIndex);
-    addField(mesh, "vy", d.vy.data(), startIndex, endIndex);
-    addField(mesh, "vz", d.vz.data(), startIndex, endIndex);
-    addField(mesh, "Mass", d.m.data(), startIndex, endIndex);
-    addField(mesh, "Smoothing Length", d.h.data(), startIndex, endIndex);
-    addField(mesh, "Density", d.rho.data(), startIndex, endIndex);
-    addField(mesh, "Internal Energy", d.u.data(), startIndex, endIndex);
-    addField(mesh, "Pressure", d.p.data(), startIndex, endIndex);
-    addField(mesh, "Speed of Sound", d.c.data(), startIndex, endIndex);
-    addField(mesh, "ax", d.ax.data(), startIndex, endIndex);
-    addField(mesh, "ax", d.ay.data(), startIndex, endIndex);
-    addField(mesh, "ax", d.az.data(), startIndex, endIndex);
-
-    std::vector<conduit_int64> conn(endIndex - startIndex);
-    std::iota(conn.begin(), conn.end(), 0);
-    mesh["topologies/mesh/elements/connectivity"].set_external(conn);
+    mesh["topologies/mesh/type"]           = "unstructured";
+    mesh["topologies/mesh/coordset"]       = "coords";
     mesh["topologies/mesh/elements/shape"] = "point";
+    std::vector<conduit_int32> conn(endIndex - startIndex);
+    for (int i = 0; i < endIndex - startIndex; i++)
+    {
+        conn[i] = i;
+    }
+
+    mesh["topologies/mesh/elements/connectivity"].set_external(conn);
+
+    mesh["fields/particles/association"] = "vertex";
+    mesh["fields/particles/topology"]    = "mesh";
+    mesh["fields/particles/values"].set_external(&d.x[startIndex], endIndex - startIndex);
+
+    std::vector<conduit_int64> ranks(endIndex - startIndex);
```

Comment:
Here I would use a 32-bit integer, MPI anyway does not support more than 2^31-1 ranks.